src/shadowsocks_config.ts

```diff
@@ -338,3 +339,63 @@ export const SIP002_URI = {
     return `ss://${userInfo}@${uriHost}:${port.data}/${queryString}${hash}`;
   },
 };
+
+export interface OnlineConfig {
+  // URL endpoint to retrieve a Shadowsocks configuration.
+  readonly url: string;
+  // Any additional configuration (e.g. `certFp`, `httpMethod`, etc.) may be stored here.
+  readonly extra: {[key: string]: string};
+}
+
+// Ref: https://github.com/shadowsocks/shadowsocks-org/issues/89
+export const SIP008_URI = {
+  PROTOCOL: 'ssconf',
+
+  validateProtocol: (uri: string) => {
+    if (!uri || !uri.startsWith(SIP008_URI.PROTOCOL)) {
```

Comment:
Take a URL object as input. Use:
url.protocol === exports.SIP008_URI.PROTOCOL

Reply:
I don't understand. Are you calling validateProtocol from native code?
We should actually just pass the protocol. It shouldn't have to know about URIs.

The current code is actually broken. It will return true for `ssconf2`. We should be handling URI parts, not as a string, which is usually troublesome.

Reply:
URL implementations behave differently across platforms. For example, older Safari webiews throw when the URL protocol is ss/ssconf; Chrome returns an empty string for the default port of a given protocol. 
By having this function take a string, we can handle these quirks internally. Otherwise, we would have to case them for each platform.

Fixed the protocol validation in c325f47, good catch.

(I think you may be looking at an earlier commit, since I removed `validateProtocol` and renamed SIP008_URI).